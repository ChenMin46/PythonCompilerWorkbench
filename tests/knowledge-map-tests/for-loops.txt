for elt in my_list:
    print elt

for elt in [1, 2, 3]:
    print elt

for elt in range(5):
    print elt

for elt in range(len(lst)):
    print elt

for (i, elt) in enumerate(lst):
    print i, elt

# multiline test
for (i, elt) \
       in \
           enumerate(lst):
    print i, elt

for x in range(100):
    if x < 2:
        break
    elif x > 10:
        continue
    else:
        print x

#for k, v in d.iteritems():
#    print k, v
#
#for ty in type(obj).mro():
#    print 'foo'
