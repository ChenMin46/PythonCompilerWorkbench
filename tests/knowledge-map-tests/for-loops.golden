for elt in my_list:
    print elt

for elt in [1, 2, 3]:
    print elt

for elt in range(5):
    print elt

for elt in range(len(lst)):
    print elt

for (i, elt) in enumerate(lst):
    print i, elt

# multiline test
for (i, elt) \
       in \
           enumerate(lst):
    print i, elt

for x in range(100):
    if x < 2:
        break
    elif x > 10:
        continue
    else:
        print x

for k, v in d.iteritems():
    print k, v

for ty in type(obj).mro():
    print 'foo'
===
Module
  body: [
    For >> 'for elt in my_list'
      target: Name >> 'elt'
        id: 'elt'
        ctx: Store
      iter: Name >> 'my_list'
        id: 'my_list'
        ctx: Load
      body: [
        Print >> 'print elt'
          dest: None
          values: [
            Name >> 'elt'
              id: 'elt'
              ctx: Load
          ]
          nl: True
      ]
      orelse: [
      ]
    For >> 'for elt in [1, 2, 3]'
      target: Name >> 'elt'
        id: 'elt'
        ctx: Store
      iter: List >> '[1, 2, 3]'
        elts: [
          Num >> '1'
            n: 1
          Num >> '2'
            n: 2
          Num >> '3'
            n: 3
        ]
        ctx: Load
      body: [
        Print >> 'print elt'
          dest: None
          values: [
            Name >> 'elt'
              id: 'elt'
              ctx: Load
          ]
          nl: True
      ]
      orelse: [
      ]
    For >> 'for elt in range(5)'
      target: Name >> 'elt'
        id: 'elt'
        ctx: Store
      iter: Call >> 'range(5)'
        func: Name >> 'range'
          id: 'range'
          ctx: Load
        args: [
          Num >> '5'
            n: 5
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
      body: [
        Print >> 'print elt'
          dest: None
          values: [
            Name >> 'elt'
              id: 'elt'
              ctx: Load
          ]
          nl: True
      ]
      orelse: [
      ]
    For >> 'for elt in range(len(lst))'
      target: Name >> 'elt'
        id: 'elt'
        ctx: Store
      iter: Call >> 'range(len(lst))'
        func: Name >> 'range'
          id: 'range'
          ctx: Load
        args: [
          Call >> 'len(lst)'
            func: Name >> 'len'
              id: 'len'
              ctx: Load
            args: [
              Name >> 'lst'
                id: 'lst'
                ctx: Load
            ]
            keywords: [
            ]
            starargs: None
            kwargs: None
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
      body: [
        Print >> 'print elt'
          dest: None
          values: [
            Name >> 'elt'
              id: 'elt'
              ctx: Load
          ]
          nl: True
      ]
      orelse: [
      ]
    For >> 'for (i, elt) in enumerate(lst)'
      target: Tuple >> '(i, elt)'
        elts: [
          Name >> 'i'
            id: 'i'
            ctx: Store
          Name >> 'elt'
            id: 'elt'
            ctx: Store
        ]
        ctx: Store
      iter: Call >> 'enumerate(lst)'
        func: Name >> 'enumerate'
          id: 'enumerate'
          ctx: Load
        args: [
          Name >> 'lst'
            id: 'lst'
            ctx: Load
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
      body: [
        Print >> 'print i, elt'
          dest: None
          values: [
            Name >> 'i'
              id: 'i'
              ctx: Load
            Name >> 'elt'
              id: 'elt'
              ctx: Load
          ]
          nl: True
      ]
      orelse: [
      ]
    For ML>> 'for (i, elt) \\\n       in \\\n           enumerate(lst)'
      target: Tuple >> '(i, elt)'
        elts: [
          Name >> 'i'
            id: 'i'
            ctx: Store
          Name >> 'elt'
            id: 'elt'
            ctx: Store
        ]
        ctx: Store
      iter: Call >> 'enumerate(lst)'
        func: Name >> 'enumerate'
          id: 'enumerate'
          ctx: Load
        args: [
          Name >> 'lst'
            id: 'lst'
            ctx: Load
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
      body: [
        Print >> 'print i, elt'
          dest: None
          values: [
            Name >> 'i'
              id: 'i'
              ctx: Load
            Name >> 'elt'
              id: 'elt'
              ctx: Load
          ]
          nl: True
      ]
      orelse: [
      ]
    For >> 'for x in range(100)'
      target: Name >> 'x'
        id: 'x'
        ctx: Store
      iter: Call >> 'range(100)'
        func: Name >> 'range'
          id: 'range'
          ctx: Load
        args: [
          Num >> '100'
            n: 100
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
      body: [
        If >> 'if x < 2'
          test: Compare >> 'x < 2'
            left: Name >> 'x'
              id: 'x'
              ctx: Load
            ops: [
              Lt
            ]
            comparators: [
              Num >> '2'
                n: 2
            ]
          body: [
            Break >> 'break'
          ]
          orelse: [
            If >> 'elif x > 10'
              test: Compare >> 'x > 10'
                left: Name >> 'x'
                  id: 'x'
                  ctx: Load
                ops: [
                  Gt
                ]
                comparators: [
                  Num >> '10'
                    n: 10
                ]
              body: [
                Continue >> 'continue'
              ]
              orelse: [
                Print >> 'print x'
                  dest: None
                  values: [
                    Name >> 'x'
                      id: 'x'
                      ctx: Load
                  ]
                  nl: True
              ]
          ]
      ]
      orelse: [
      ]
    For >> 'for k, v in d.iteritems()'
      target: Tuple >> 'k, v'
        elts: [
          Name >> 'k'
            id: 'k'
            ctx: Store
          Name >> 'v'
            id: 'v'
            ctx: Store
        ]
        ctx: Store
      iter: Call >> 'd.iteritems()'
        func: Attribute >> 'd.iteritems'
          value: Name >> 'd'
            id: 'd'
            ctx: Load
          attr: 'iteritems'
          ctx: Load
        args: [
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
      body: [
        Print >> 'print k, v'
          dest: None
          values: [
            Name >> 'k'
              id: 'k'
              ctx: Load
            Name >> 'v'
              id: 'v'
              ctx: Load
          ]
          nl: True
      ]
      orelse: [
      ]
    For >> 'for ty in type(obj).mro()'
      target: Name >> 'ty'
        id: 'ty'
        ctx: Store
      iter: Call >> 'type(obj).mro()'
        func: Attribute >> 'type(obj).mro'
          value: Call >> 'type(obj)'
            func: Name >> 'type'
              id: 'type'
              ctx: Load
            args: [
              Name >> 'obj'
                id: 'obj'
                ctx: Load
            ]
            keywords: [
            ]
            starargs: None
            kwargs: None
          attr: 'mro'
          ctx: Load
        args: [
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
      body: [
        Print >> "print 'foo'"
          dest: None
          values: [
            Str >> "'foo'"
              s: 'foo'
          ]
          nl: True
      ]
      orelse: [
      ]
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "id": "tid_1", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " ", 
            {
              "id": "tid_2", 
              "name": "token", 
              "value": "in"
            }, 
            " ", 
            {
              "contents": [
                "my_list"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_3", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_4", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_5", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " ", 
            {
              "id": "tid_6", 
              "name": "token", 
              "value": "in"
            }, 
            " ", 
            {
              "contents": [
                {
                  "id": "tid_7", 
                  "name": "token", 
                  "value": "["
                }, 
                {
                  "contents": [
                    "1"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_8", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "2"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_9", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "3"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_10", 
                  "name": "token", 
                  "value": "]"
                }
              ], 
              "id": "id_0", 
              "name": "List"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_11", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_12", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_13", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " ", 
            {
              "id": "tid_14", 
              "name": "token", 
              "value": "in"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    "range"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_15", 
                  "name": "token", 
                  "value": "("
                }, 
                {
                  "contents": [
                    "5"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_16", 
                  "name": "token", 
                  "value": ")"
                }
              ], 
              "id": "id_0", 
              "name": "Call"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_17", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_18", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_19", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " ", 
            {
              "id": "tid_20", 
              "name": "token", 
              "value": "in"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    "range"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_21", 
                  "name": "token", 
                  "value": "("
                }, 
                {
                  "contents": [
                    {
                      "contents": [
                        "len"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    {
                      "id": "tid_22", 
                      "name": "token", 
                      "value": "("
                    }, 
                    {
                      "contents": [
                        "lst"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    {
                      "id": "tid_23", 
                      "name": "token", 
                      "value": ")"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "Call"
                }, 
                {
                  "id": "tid_24", 
                  "name": "token", 
                  "value": ")"
                }
              ], 
              "id": "id_0", 
              "name": "Call"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_25", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_26", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_27", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                {
                  "id": "tid_28", 
                  "name": "token", 
                  "value": "("
                }, 
                {
                  "contents": [
                    "i"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_29", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "elt"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_30", 
                  "name": "token", 
                  "value": ")"
                }
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }, 
            " ", 
            {
              "id": "tid_31", 
              "name": "token", 
              "value": "in"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    "enumerate"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_32", 
                  "name": "token", 
                  "value": "("
                }, 
                {
                  "contents": [
                    "lst"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_33", 
                  "name": "token", 
                  "value": ")"
                }
              ], 
              "id": "id_0", 
              "name": "Call"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_34", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_35", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "i"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            {
              "id": "tid_36", 
              "name": "token", 
              "value": ","
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "", 
        "\n", 
        "", 
        "\n", 
        "", 
        {
          "id": "tid_37", 
          "name": "comment", 
          "value": "# multiline test"
        }, 
        "\n", 
        "", 
        {
          "contents": [
            {
              "id": "tid_38", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                {
                  "id": "tid_39", 
                  "name": "token", 
                  "value": "("
                }, 
                {
                  "contents": [
                    "i"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_40", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "elt"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_41", 
                  "name": "token", 
                  "value": ")"
                }
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }, 
            " ", 
            {
              "id": "tid_42", 
              "name": "token", 
              "value": "\\"
            }, 
            "\n", 
            "       in \\", 
            "\n", 
            "           ", 
            {
              "contents": [
                {
                  "contents": [
                    "enumerate"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_43", 
                  "name": "token", 
                  "value": "("
                }, 
                {
                  "contents": [
                    "lst"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_44", 
                  "name": "token", 
                  "value": ")"
                }
              ], 
              "id": "id_0", 
              "name": "Call"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_45", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_46", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "i"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            {
              "id": "tid_47", 
              "name": "token", 
              "value": ","
            }, 
            " ", 
            {
              "contents": [
                "elt"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_48", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                "x"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " ", 
            {
              "id": "tid_49", 
              "name": "token", 
              "value": "in"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    "range"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_50", 
                  "name": "token", 
                  "value": "("
                }, 
                {
                  "contents": [
                    "100"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_51", 
                  "name": "token", 
                  "value": ")"
                }
              ], 
              "id": "id_0", 
              "name": "Call"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_52", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_53", 
              "name": "token", 
              "value": "if"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    "x"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                " ", 
                {
                  "id": "tid_54", 
                  "name": "token", 
                  "value": "<"
                }, 
                " ", 
                {
                  "contents": [
                    "2"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }
              ], 
              "id": "id_0", 
              "name": "Compare"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "If"
        }, 
        {
          "id": "tid_55", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "        ", 
        {
          "contents": [
            {
              "id": "tid_56", 
              "name": "token", 
              "value": "break"
            }
          ], 
          "id": "id_0", 
          "name": "Break"
        }, 
        "\n    ", 
        {
          "contents": [
            {
              "id": "tid_57", 
              "name": "token", 
              "value": "elif"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    "x"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                " ", 
                {
                  "id": "tid_58", 
                  "name": "token", 
                  "value": ">"
                }, 
                " ", 
                {
                  "contents": [
                    "10"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }
              ], 
              "id": "id_0", 
              "name": "Compare"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "If"
        }, 
        {
          "id": "tid_59", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "        ", 
        {
          "contents": [
            {
              "id": "tid_60", 
              "name": "token", 
              "value": "continue"
            }
          ], 
          "id": "id_0", 
          "name": "Continue"
        }, 
        "", 
        "\n", 
        "    else:", 
        "\n", 
        "        ", 
        {
          "contents": [
            {
              "id": "tid_61", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "x"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_62", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    "k"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_63", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "v"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }
              ], 
              "id": "id_0", 
              "name": "Tuple"
            }, 
            " ", 
            {
              "id": "tid_64", 
              "name": "token", 
              "value": "in"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    {
                      "contents": [
                        "d"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    ".iteritems"
                  ], 
                  "id": "id_0", 
                  "name": "Attribute"
                }, 
                "()"
              ], 
              "id": "id_0", 
              "name": "Call"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_65", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_66", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "k"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            {
              "id": "tid_67", 
              "name": "token", 
              "value": ","
            }, 
            " ", 
            {
              "contents": [
                "v"
              ], 
              "id": "id_0", 
              "name": "Name"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_68", 
              "name": "token", 
              "value": "for"
            }, 
            " ", 
            {
              "contents": [
                "ty"
              ], 
              "id": "id_0", 
              "name": "Name"
            }, 
            " ", 
            {
              "id": "tid_69", 
              "name": "token", 
              "value": "in"
            }, 
            " ", 
            {
              "contents": [
                {
                  "contents": [
                    {
                      "contents": [
                        {
                          "contents": [
                            "type"
                          ], 
                          "id": "id_0", 
                          "name": "Name"
                        }, 
                        {
                          "id": "tid_70", 
                          "name": "token", 
                          "value": "("
                        }, 
                        {
                          "contents": [
                            "obj"
                          ], 
                          "id": "id_0", 
                          "name": "Name"
                        }, 
                        {
                          "id": "tid_71", 
                          "name": "token", 
                          "value": ")"
                        }
                      ], 
                      "id": "id_0", 
                      "name": "Call"
                    }, 
                    ".mro"
                  ], 
                  "id": "id_0", 
                  "name": "Attribute"
                }, 
                "()"
              ], 
              "id": "id_0", 
              "name": "Call"
            }, 
            ""
          ], 
          "id": "id_0", 
          "name": "For"
        }, 
        {
          "id": "tid_72", 
          "name": "token", 
          "value": ":"
        }, 
        "\n", 
        "    ", 
        {
          "contents": [
            {
              "id": "tid_73", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "contents": [
                "'foo'"
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
